tests/test_config.py

```diff
@@ -0,0 +1,141 @@
+import sys
+import os
+import pathlib
+import pytest
+
+_rundir = pathlib.Path(__file__).parent
+if not str(_rundir.parent) in sys.path:
+    sys.path.insert(0, str(_rundir.parent ) )
+from util import config
+
+# A note about pytest: Things aren't completely sandboxed.  When I call
+# config.Config.get(), it selts Config._default, and that carries over
+# from one test to the next even if the call wasn't in a fixture with
+# class scope.  (The tests below are ordered with this in mind.)
+
+class TestConfig:
+    def test_no_default( self ):
+        assert config.Config._default == None
+
+    @pytest.mark.skipif( os.getenv("SEECHANGE_CONFIG") is None, reason="Set SEECHANGE_CONFIG to test this" )
+    def test_default_default( self ):
+        cfg = config.Config.get()
+        assert cfg._path == pathlib.Path( os.getenv("CURVEBALL_CONFIG") )
+
+    def test_set_default( self ):
+        cfg = config.Config.get( _rundir / 'test.yaml', setdefault=True )
+        assert config.Config._default == f'{ (_rundir / "test.yaml").resolve() }'
+
+    @pytest.fixture(scope='class')
+    def cfg( self ):
+        return config.Config.get( _rundir / 'test.yaml', setdefault=True )
```

Comment:
This may be part of a larger style conversation, but:

I don't think it's worth having a whole separate file for three lines of code.  I could see the argument that the fixtures should be at the top.  And, if there are several fixtures, then, yes, I can see the reason to have them in a separate file.  But a quick thing that a file needs that is only three lines long, doing it for the sake of consistency isn't really worth it, and if anything chops the code up enough to make it harder to read later.   I don't see this fixture being used anywhere else (since the test yaml file is something designed just for this test, and won't be useful elsewhere).

Re: the larger conversation, I am going to object strenuously to rigid style rules.  To my mind, the very most important part of PEP-8 is the first section after the Introduction, and I think the writers of PEP-8 were smart in making that the first thing.  (There are some other parts of PEP-8 I would diagree with, but I think it's more important in that case just not to bother getting into arguments about it.)  I know some people think you should use tools like "black" on your codebase, but I hate that.  (Black in particular destroys so much readability of the code that it hinders maintainability.   As I inherited Lensgrinder from Danny, at one point early on he ran black on the entire codebase.  I spent a lot of time thereafter removing the ugliness that that one change had imposed.)

Reply:
This test doesn't have a lot of (reusable) fixtures, but down the line we will have many. 
The question is this: do we want to keep the "relevant for this test only" fixtures in the same file as the test, and only have the "shared" fixtures in `conftest.py` or should we just put all of them in one place so it is less work to find them? 

I don't mind either way, but we will certainly have a `conftest.py` file.